File: solutions/kotlin/raindrops/1/src/main/kotlin/Raindrops.kt
object Raindrops {
    val soundMapping = arrayOf(3 to "Pling", 5 to "Plang", 7 to "Plong")
    fun convert(n: Int): String =
        soundMapping.filter{ (divisor, _) ->
            n % divisor == 0
        }.map{ (_, sound) ->
            sound
        }.joinToString(separator = "").let{ result ->
            if (result.isEmpty()) n.toString() else result
        }
}

File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {
    std::string convert(int drops) {
        std::string res{};
        if (drops % 3 == 0) {
            res += "Pling";
        }
        if (drops % 5 == 0) {
            res += "Plang";
        }
        if (drops % 7 == 0) {
            res += "Plong";
        }
        if (res.size() == 0) {
            res += std::to_string(drops);
        }
        return res;
    }
}
